lib/filewatch/watched_files_collection.rb

```diff
@@ -15,13 +15,16 @@ def add(watched_file)
       @sort_method.call
     end
 
-    def delete(paths)
-      Array(paths).each do |f|
-        index = @pointers.delete(f)
-        @files.delete_at(index)
-        refresh_pointers
+    def remove_paths(paths)
+      removed_files = Array(paths).map do |path|
+        if index = @pointers.delete(path)
+          watched_file = @files.delete_at(index)
+          refresh_pointers
+          watched_file
+        end
       end
       @sort_method.call
+      removed_files
```

Comment:
It's not clear to me why we the `remove_paths` returns the list of removed files. The method `remove_paths` is used inside the `DeleteCompletedFileHandler.handle` which on its own return the list of deleted files. That method is invoked by `File.handle_deletable_path` in an each loop and the method itself doesn't return nothing.

Reply:
Thanks Andrea, might have been a bit ahead of myself here ...
It's common for a (map) remove operation to return removed values.
The watched files collection is a map (`[String] path -> [WatchedFile] file`), wanted to re-turn smt meaningful (planning to spec the return value with a collection rewrite to native).